lib/spack/spack/build_systems/meson.py

```diff
@@ -55,14 +55,15 @@ class MesonPackage(PackageBase):
     variant('buildtype', default='debugoptimized',
             description='Meson build type',
             values=('plain', 'debug', 'debugoptimized', 'release', 'minsize'))
-    variant('default_library', default='shared',
-            description=' Default library type',
-            values=('shared', 'static', 'both'))
+    variant('shared', default=True, description='Build shared libraries')
+    variant('static', default=False, description='Build static libraries')
```

Comment:
Per discussion in #23474 I think we should turn this one into a multi-valued variant.